pkg/cvo/upgradeable.go

```diff
@@ -442,3 +449,31 @@ func (optr *Operator) adminGatesEventHandler() cache.ResourceEventHandler {
 		DeleteFunc: optr.deleteFunc,
 	}
 }
+
+// shouldSyncUpgradeableDueToPreconditionChecks checks if the upgradeable status should
+// be synchronized due to the precondition checks. It checks whether the precondition
+// checks on the payload are failing for less than minimumUpdateCheckInterval, and it has
+// been more than the statusInterval since the last synchronization. This means we will
+// synchronize the upgradeable status more frequently at beginning of an upgrade for
+// a while in a case the precondition checks are failing.
```

Comment:
> // ...This means we will
> // synchronize the upgradeable status more frequently at beginning of an upgrade for
> // a while in a case the precondition checks are failing.
How about just:
`This means, upon precondition failure, we will synchronize the upgradeable status more frequently at beginning of an upgrade.`

Reply:
Thanks for the better wording, Jack! 
Comments replaced in the new commit https://github.com/openshift/cluster-version-operator/commit/5aac7a3ef2934cbeae09a2632e22fe57eb20da91.